flang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -859,6 +859,12 @@ static void parsePreprocessorArgs(Fortran::frontend::PreprocessorOptions &opts,
         (currentArg->getOption().matches(clang::driver::options::OPT_cpp))
             ? PPMacrosFlag::Include
             : PPMacrosFlag::Exclude;
+  // Enable -cpp based on -x unless explicitly disabled with -nocpp
+  if (const auto *dashX = args.getLastArg(clang::driver::options::OPT_x);
+      dashX && opts.macrosFlag != PPMacrosFlag::Exclude)
```

Comment:
I missed this the first time around. Is there precedent in the LLVM project for multiple statements in conditional expressions? Might it be cleaner to just use nested conditionals?

```suggestion
if (opts.macrosFlag != PPMacrosFlag::Exclude)
  if (const auto *dashX = args.getLastArg(clang::driver::options::OPT_x)
```

Reply:
Done